components/Auctions/createNewAuction.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect } from 'react';
+import { useRouter } from 'next/router';
+import styles from './new.module.css';
+import fetchData from '../../utils/fetchData';
+import fetchSelfDetails from '../../utils/fetchSelfDetails';
+
+const BASE_API_URL = process.env.NEXT_PUBLIC_BASE_API_URL;
+const AUCTIONS_URL = `${BASE_API_URL}/auctions`;
+
+const CreateNewAuction = () => {
+  const router = useRouter();
+
+  const [itemType, setItemType] = useState();
+  const [quantity, setQuantity] = useState();
+  const [initialPrice, setInitialPrice] = useState();
+  const [duration, setDuration] = useState();
+  const [isSubmitted, setIsSubmitted] = useState(false);
+
+  useEffect(() => {
+    fetchSelfDetails()
+      .then((res) => {
+        if (res.status === 401) {
+          alert('You need to be logged in to be able to create an auction!');
+        }
+      })
+      .catch((err) => {
+        console.log('Some error occured', err);
+      });
+  }, []);
+
+  const auctionSubmitHandler = async (e) => {
+    e.preventDefault();
+    setIsSubmitted(true);
+    const reqBody = {
+      item_type: itemType,
+      initial_price: initialPrice,
+      duration,
+      quantity,
+    };
+
+    const response = await fetchData(`${AUCTIONS_URL}`, 'POST', {
+      credentials: 'include',
+      body: JSON.stringify(reqBody),
+    });
+    const { status } = response;
+    if (status == 204) {
+      setIsSubmitted(false);
+      router.push('/auctions');
+    }
+  };
+
+  return (
+    <div className={styles.formContainer}>
+      <form className={styles.newAuctionForm} onSubmit={auctionSubmitHandler}>
+        {isSubmitted && <p>Please wait...</p>}
+        <label name="itemType">
+          Item to auction:
+          <br />
+          <input
+            className={styles.inputBox}
+            htmlFor="itemType"
+            onChange={(e) => setItemType(e.target.value)}
+            required
```

Comment:
the `htmlFor` should be used in label right? and there you need to pass the reference of this input(using id)